Add support of SELinux and Apparmor

surajssd opened the pull request:
* Add support to disable SELinux using a parameter `selinux`. It has two inputs `enabled` and `disabled`. Default behavior of the containers is with SELinux enalbed.
    
* Add support to disable Apparmor using a parameter `apparmor`. It has two inputs `enabled` and `disabled`. Default behavior of the containers is with Apparmor enalbed.
    
* Add flag `-os` to specify what OS the application container is running on.

* Depending on what OS the app runs on the index.html is generated by temeplating. So for fedora on SELinux regarding dropdown will be shown and for ubuntu Apparmor dropdown will be shown.

* Removed the static index.html

surajssd committed:
api: Add support to disable selinux

Add support to disable SELinux using a parameter `selinux`. It has two
inputs `enabled` and `disabled`. Default behavior of the containers is
with SELinux enalbed.

Signed-off-by: Suraj Deshmukh <suraj@kinvolk.io>

surajssd committed:
frontend: Support for the SELinux dropdown

Enable or Disable SELinux using the drop down.

Signed-off-by: Suraj Deshmukh <suraj@kinvolk.io>

surajssd commented:
### Testing this PR

Deploy this application, and once you get to the profiles page put any docker image and don't change anything else. Once you get the prompt then run following commands

```
cd /var/tmp/shared
touch from-container
```

If you get `Permission Denied` that means SELinux is enforcing.

---

To test it with disabled SELinux run the same set of commands and you should not be getting any errors.

pothos commented:
As the apparmor and selinux PRs touch the same files at the same location, wouldn't it be better to merge them into one so that we also test the combined effects (and don't have merge conflicts)?

pothos commented on frontend/index.html:
```diff
@@ -37,6 +37,14 @@ <h1>Profiles</h1>
+            <select name="selinux">
+                <option value="enabled">Enabled</option>
+                <option value="disabled">Disabled</option>
+            </select>
```
Does not have to be in this PR, but it would be good if the Fedora VM only shows this while the Ubuntu VM only shows AppArmor. Maybe it's possible to pass a parameter to the provisioning script so that only one is shown?

surajssd replied:
This can be done we generate this on the fly. Right now this is statically generated. Will start working on that.

surajssd replied:
this is done.

surajssd committed:
api: Add support to disable apparmor

Add support to disable Apparmor using a parameter `apparmor`. It has two
inputs `enabled` and `disabled`. Default behavior of the containers is
with Apparmor enalbed.

Signed-off-by: Suraj Deshmukh <suraj@kinvolk.io>

surajssd committed:
frontend: Support for the Apparmor dropdown

Enable or Disable Apparmor using the drop down.

Signed-off-by: Suraj Deshmukh <suraj@kinvolk.io>

surajssd commented:
@pothos 

> wouldn't it be better to merge them into one so that we also test the combined effects (and don't have merge conflicts)?

pulled the commits from other PR.

surajssd committed:
cli: Add new flag `-os`

This flag will let you specify the Operating System of docker host.

Signed-off-by: Suraj Deshmukh <suraj@kinvolk.io>

surajssd committed:
OS specific index.html

This commit generates index.html file on the go depending on what
Operating System the provided `-os` flag says.

If the OS is `fedora` then the SELinux regarding dropdown is put in the
index.html file, else if it is `ubuntu` then the dropdown is for
Apparmor.

Signed-off-by: Suraj Deshmukh <suraj@kinvolk.io>

surajssd committed:
frontend: Remove static index.html

Signed-off-by: Suraj Deshmukh <suraj@kinvolk.io>
